src/main.js

```diff
@@ -1,23 +1,25 @@
+// @flow
+
 import { getDate, getDayOfYear, getDaysInMonth, getDaysInYear, isMonday, isSameDay, isSunday, subDays, subYears } from './helpers';
 import { applyOffset, formatDate, getDiffFn, getEndOfFn, getStartOfFn, getSubtractionFn } from './utils';
+import type { CompareMode, DateLike, Period, PeriodParams, Range, RangeAndLabel, Unit } from './types.js.flow';
 
-// :: (Date | Int | String) -> (Date | Int | String)  -> Object
-const formatRange = (start, end) => ({
+const formatRange = (start: DateLike, end: DateLike): Range => ({
   start: formatDate(start),
   end: formatDate(end),
 });
 
-// :: Int -> String -> Option(Date | Int | String) = new Date() -> Object
-const getLastRelativePeriodRange = (num, unit, base = new Date()) => {
+const getLastRelativePeriodRange = (num: number, unit: Unit, base: Date = new Date()): Range => {
```

Comment:
Add `?` to `base` as it is optional?

And should `base` be `DateLike` instead?